### PR TITLE
[Yaml] removed YAML parser \ escaping in double-quotes

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Should be rebased when #16201 is merged.

### CODE REVIEW

File: src/Symfony/Component/Yaml/Unescaper.php [hunk 1]
```diff
@@ -109,6 +114,8 @@ public function unescapeCharacter($value)
                 return self::utf8chr(hexdec(substr($value, 2, 4)));
             case 'U':
                 return self::utf8chr(hexdec(substr($value, 2, 8)));
+            default:
+                throw new ParseException(sprintf('Found unknown escape character (%s in \\%s).', $value, $value));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Yaml/Unescaper.php]: in case `'' === $value`, the message could be enhanced imo

Comment (fabpot) [inline src/Symfony/Component/Yaml/Unescaper.php]: As mentioned in https://github.com/symfony/symfony/pull/16201/files#r41711538, this is not possible.

**Comment by fabpot** [comment]
rebased

File: src/Symfony/Component/Yaml/Unescaper.php [hunk 2]
```diff
@@ -63,7 +65,7 @@ public function unescapeDoubleQuotedString($value)
      * @internal This method is public to be usable as callback. It should not
```
Comment (aitboudad) [inline src/Symfony/Component/Yaml/Unescaper.php]: you should remove the `internal` tag too

Comment (fabpot) [inline src/Symfony/Component/Yaml/Unescaper.php]: right, done

**Comment by nicolas-grekas** [comment]
:+1:

**Pull request merged**